finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/PathMatcher.scala

```diff
@@ -75,18 +75,29 @@ object PathMatcher {
             _extract(pathSegments.tail, exprSegments.tail, vars)
           else if (exprSegment == pathSegment)
             _extract(pathSegments.tail, exprSegments.tail, vars)
-          else if (exprSegment.startsWith("{") && exprSegment.endsWith("}"))
-            _extract(
-              pathSegments.tail,
-              exprSegments.tail,
-              vars + (exprSegment.substring(1, exprSegment.length - 1) -> pathSegment)
-            )
           else
-            None
+            """\{([a-zA-Z0-9\\.:-]+)\}""".r.findAllIn(exprSegment) match {
+              case k if !k.isEmpty && pathSegment.startsWith(exprSegment.take(k.start)) =>
+                var keys = k.toArray
+                keys.foldLeft(exprSegment.replace(".", "\\."))(_re).r.findAllIn(pathSegment) match {
```

Comment:
Just to confirm that I understand, we only escape "." because that's the only valid Path character which also has a regex meaning?

Reply:
Correct.